app/main.py

```diff
@@ -1,2 +1,63 @@
+from dataclasses import dataclass
+from typing import Hashable, Any
+
+
+@dataclass
+class Node:
+    key: Hashable
+    value: any
+    key_hash: int
+
+
 class Dictionary:
-    pass
+    INITIAL_CAPACITY = 8
+    THRESHOLD = 2 / 3
+    CAPACITY_MULTIPLIER = 2
+
+    def __init__(self, capacity: int = INITIAL_CAPACITY) -> None:
+        self.capacity = capacity
+        self.size = 0
+        self.hash_table = [None] * self.capacity
+
+    def index_change(self, index: int) -> int:
+        return (index + 1) % self.capacity
+
+    def calculate_index(self, key: Hashable) -> int:
+        key_hash = hash(key)
+        index = key_hash % self.capacity
+        while (node := self.hash_table[index]) is not None:
+            if node.key == key and node.key_hash == key_hash:
+                break
+            index = self.index_change(index)
+        return index
+
+    def resize(self) -> None:
+        self.capacity *= self.CAPACITY_MULTIPLIER
+        old_hash_table = self.hash_table
+        self.hash_table = [None] * self.capacity
+        self.size = 0
```

Comment:
Resetting `self.size` to 0 during resizing is incorrect. The size should reflect the number of elements in the dictionary, which does not change during a resize operation. Consider removing this line.